src/lib.rs

```diff
@@ -450,8 +447,38 @@ mod tests {
         )
         .unwrap();
 
-        let bytes = img.save_to_bufferv("png", &[]).unwrap();
+        let target = image::open("data/wikipedia_logo_blurred.png").unwrap();
+        assert_image_data_approximately_equal(&img.read_pixel_bytes(), target.as_bytes())
+    }
+
+    #[cfg(feature = "gdk-pixbuf")]
+    fn assert_image_data_approximately_equal(result: &[u8], target: &[u8]) {
+        const MAX_AVERAGE_ERROR: usize = 1;
+        const MAX_PEAK_ERROR: usize = 8;
+
+        if result.len() != target.len() {
+            panic!(
+                "images do not have the same shape: {} vs {}",
+                result.len(),
+                target.len()
+            );
+        }
 
-        assert_eq!(bytes[1000..1005], [77, 210, 4, 80, 15]);
+        let mut aggregated_error: usize = 0;
+        let mut peak_error = 0;
+        for (r, t) in result.iter().zip(target) {
+            let error = (*r as isize - *t as isize).abs() as usize;
+            aggregated_error += error;
+            peak_error = peak_error.max(error);
+        }
+
+        let average_error = aggregated_error / result.len();
+
+        if average_error > MAX_AVERAGE_ERROR || peak_error > MAX_PEAK_ERROR {
+            panic!(
+                "images do not look similar: average error {}; peak error {}",
+                average_error, peak_error
+            );
+        }
```

Comment:
```suggestion
        assert!(average_error <= MAX_AVERAGE_ERROR, "images do not look similar. average error {average_error} > {MAX_AVERAGE_ERROR}");
        assert!(peak_error <= MAX_PEAK_ERROR, "images do not look similar. peak error {peak_error} > {MAX_PEAK_ERROR}");
```

I think asserts are a bit more clear in this context. Maybe the same above.